lib/msf/core/exploit/remote/kerberos/client.rb

```diff
@@ -155,8 +155,8 @@ def send_request_tgt(options = {})
             server_name = options[:server_name]
             client_name = options[:client_name]
             password = options[:password]
-            password.force_encoding('utf-8') if password
-            client_name.force_encoding('utf-8')
+            password.dup.force_encoding('utf-8') if password
+            client_name.dup.force_encoding('utf-8')
```

Comment:
Confirmed this fixes the password logging with diacritics and rc4 enc type (I'll get rid of the `.inspect` in a separate PR):

> [+] 192.168.123.13 - User found: "w\xC3\xAEth.d\xC3\xAC\xC3\xA1\xC3\xA7ritic\xC5\xA1" with password wîth.dìáçriticš. Hash: $krb5asrep$18$wîth.dìáçriticš@ADF3.LOCAL:6f4c32ba20d1ae336cb78b8ae2f3c9d3$c0faa44a6e753f7282ff08fa0049514454a31e1162a7d20d80bd35ff9535ce246e20d4cc91ced03e2685971502e3b96a192d3e43f08dab0cfe81031d4c5d5f268b7c4e56a2d4e20212c7114980e1fdd875836b1f2803fc02f1b3e3f19d8c6154b2583af02fe0bc13dc8618e0f86a86cb22079c75192043545f29d24b86a56ba74f8a869b31977e43bbbd527cc0524785bfaee7a22e5e2954faa5e2cf1a940b8a78ba6423fd22dc970d32706171a2967f265701d2cff62060dec8ed29770cc6e41ebf5a59be6a8fbdb0453b8ce2896b7ac85742380756ca4277d257d6f2d4adb9c2a4ed376de967a098e73377e86d0617e1cf68cf5b45b8f09eca97